lib/src/main/kotlin/org/schedo/scheduler/SchedulerBuilder.kt

```diff
@@ -35,23 +38,30 @@ class SchedulerBuilder {
 
     fun build(): Scheduler {
         val dsType = dataSourceType  // snapshot it
-        when (dsType) {
-            null -> {}
-            is DataSourceType.Postgres -> createPostgresTables(dataSource!!)
-            is DataSourceType.Other -> TODO("${dsType.name} is not supported")
-        }
 
-        val tasksRepository: TasksRepository = when (dsType) {
-            null -> InMemoryTasks()
-            is DataSourceType.Postgres -> PostgresTasksRepository(dataSource!!)
-            is DataSourceType.Other -> TODO()
-        }
-        val statusRepository: StatusRepository = when (dsType) {
-            null -> InMemoryStatus()
-            is DataSourceType.Postgres -> PostgresStatusRepository(dataSource!!)
-            is DataSourceType.Other -> TODO()
+        val (tasksRepository, statusRepository, retryRepository) = when (dsType) {
+            null -> {
+                val inMemTasks  = InMemoryTasks()
+                val inMemStatus = InMemoryStatus()
+                Triple<TasksRepository, StatusRepository, RetryRepository>(
+                    inMemTasks,
+                    inMemStatus,
+                    InMemoryRetry(inMemTasks, inMemStatus)
+                )
+            }
+
+            is DataSourceType.Postgres -> {
+                createPostgresTables(dataSource!!)
+                val pgTasks  = PostgresTasksRepository(dataSource!!)
+                val pgStatus = PostgresStatusRepository(dataSource!!)
+                val pgRetry  = PostgresRetryRepository(dataSource!!)
+                Triple(pgTasks, pgStatus, pgRetry)
```

Comment:
Лучше не использовать `!!`, даже когда очевидно, что ошибок не будет. 

```suggestion
                val pgDataSource = checkNotNull(dataSource) { "Postgres data source is not set" }
                
                createPostgresTables(pgDataSource)
                val pgTasks  = PostgresTasksRepository(pgDataSource)
                val pgStatus = PostgresStatusRepository(pgDataSource)
                val pgRetry  = PostgresRetryRepository(pgDataSource)
                Triple(pgTasks, pgStatus, pgRetry)
```